portfolio/src/main/webapp/comments.js

```diff
@@ -82,3 +82,33 @@ async function getComments() {
     displayList.appendChild(createListElement(review));
   });
 }
+
+// Display any potential error that comes from backend
+function displayErrors() {
+  const url_string = window.location.href; //window.location.href
+  const url = new URL(url_string);
+  const error = url.searchParams.get("error");
+
+  if (error === "incorrect-name") {
+    Swal.fire({
+        icon: 'error',
+        title: 'Name Error',
+        text: 'Your name contains characters that are not allowed.',
```

Comment:
Telling the user that they used characters that weren't allowed without telling them which characters are in fact allowed would force the user to guess. Ideally, error messages allow the user to self-diagnose. Though a guessing game here is surely also fun :-).